astropy/nddata/ccddata.py

```diff
@@ -738,6 +682,66 @@ def fits_ccddata_reader(
     return ccd_data
 
 
+# this docstring *must* be available at runtime to support `UnifiedReadWrite.help`
+# so we attach it manually, which is robust against PYTHONOPTIMIZE=2
+fits_ccddata_reader.__doc__ = """
```

Comment:
A quick google search didn't reveal anything about being able to make a decorator that prevents the docstring from being removed. However if the optimizer is just doing static AST parsing then perhaps a simple decorator would work.

```
@keep_doc("""docstring""")
def function()
```

Worth a try? Keeps the docs and function more closely associated.

Reply:
I don't have a strong opinion, but it's certainly feasible. I'll push an implementation of this in a minute, let me know what you think !

Reply:
Does this works with the optimizer flag?

Reply:
It does. I should have said that when I pushed, but I did test it.